pages/admin/courses/_id.vue

```diff
@@ -0,0 +1,115 @@
+<template>
+  <div class="container">
+    <form>
+      <vue-form-generator
+        :schema="schema"
+        :model="model"
+        :options="formOptions"
+      >
+      </vue-form-generator>
+    </form>
+  </div>
+</template>
+ 
+<script>
+import Vue from 'vue'
+const VueFormGenerator = require('vue-form-generator')
+import 'vue-form-generator/dist/vfg.css'
+
+Vue.use(VueFormGenerator)
+
+export default Vue.extend({
+  name: 'CourseEditPage',
+  layout: 'admin',
+  data() {
+    return {
+      model: {},
+      schema: {
+        groups: [
+          {
+            legend: 'Course Details',
+            fields: [
+              {
+                type: 'input',
+                inputType: 'text',
+                label: 'Course Name',
+                model: 'title',
+                placeholder: 'Enter course name',
+                help: 'This will be the title for the course',
+                required: true,
+              },
+              {
+                type: 'textArea',
+                inputType: 'text',
+                label: 'Course description',
+                model: 'description',
+                placeholder: 'Enter course name',
+                help: 'This will be the summary for the course',
+                required: true,
+              },
+              {
+                type: 'input',
+                inputType: 'text',
+                label: 'Slug',
+                model: 'slug',
+                help: `This will be the <a href="https://stackoverflow.com/questions/19335215/what-is-a-slug"> slug </a> for the course.
+                  Use kebab-case (hyphen separated) name and avoid special characters if you can) For example: c++-for-beginners`,
+                placeholder: 'This will be the slug for the course',
+                required: true,
+              },
+              {
+                type: 'textArea',
```

Comment:
course syllabus is a JSON object and will have a complex editor, we can either skip this as of now or create a component that can edit different fields in the syllabus starting with just the title